mitodl/open-discussions
# Added factories for reddit objects

#### What are the relevant tickets?
Fixes #317

Adds factories for reddit objects and utilities to store and use generated factory data.

#### How should this be manually tested?
Run the factories and be sure that they create matching objects in reddit.

Example:
```python
from channels.api import Api
from django.contrib.auth.models import User
from channels.factories import *

u = User.objects.first()
api = Api(u)

channel = ChannelFactory.create(channel_type='public', api=api)
post = TextPostFactory.create(channel=channel, api=api)
comment = CommentFactory.create(post_id=post.id, api=api)
```

Then go over to `reddit.local//comments/{post.id}` and ensure it is in the expected channel with comments.

Also, delete the cassette and factory_data for one of the updated tests and be sure you can run the tests and it generates new files and a second run of the tests passes too.
